### PR TITLE
[FLINK-19147] Support CliClient AutoClose interface

### PR DESCRIPTION
**Pull request by darionyaphet**
## What is the purpose of the change

Since JDK7, using `try-with-resources` to process exceptions. When the resource implement `AutoClose` interface it could release automatic. 

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by aljoscha** [comment]
@twalthr This seems like a legit improvement, right?

@darionyaphet The build is failing because of problems with checkstyle: https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=6270&view=logs&j=52b61abe-a3cc-5bde-cc35-1bbe89bb7df5&t=54421a62-0c80-5aad-3319-094ff69180bb&l=13354

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java [hunk 1]
```diff
@@ -144,18 +138,13 @@ public void testUseNonExistingCatalog() throws Exception {
...
 		InputStream inputStream = new ByteArrayInputStream("use catalog cat;\n".getBytes());
-		CliClient cliClient = null;
 		SessionContext sessionContext = new SessionContext("test-session", new Environment());
 		String sessionId = executor.openSession(sessionContext);
 
-		try (Terminal terminal = new DumbTerminal(inputStream, new MockOutputStream())) {
-			cliClient = new CliClient(terminal, sessionId, executor, File.createTempFile("history", "tmp").toPath());
-			cliClient.open();
+		try (Terminal terminal = new DumbTerminal(inputStream, new MockOutputStream());
+			 CliClient client = new CliClient(terminal, sessionId, executor, File.createTempFile("history", "tmp").toPath())) {
```
Comment (twalthr) [inline flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java:145]: nit: fix indention here and below, seems like a typical copy paste mistake. how about we create a method to reduce duplicate code (e.g. for the tmp file)?

**Review by twalthr — CHANGES_REQUESTED** [review]
Looks like a valid improvement. I just had one comment.

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java (hunk 1 above)
Comment (darionyaphet) [reply flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java]: > nit: fix indention here and below, seems like a typical copy paste mistake. how about we create a method to reduce duplicate code (e.g. for the tmp file)?

Maybe Idea's indention is using space. Any config could change to tab?

**Comment by darionyaphet** [comment]
> @twalthr This seems like a legit improvement, right?
> 
> @darionyaphet The build is failing because of problems with checkstyle: https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=6270&view=logs&j=52b61abe-a3cc-5bde-cc35-1bbe89bb7df5&t=54421a62-0c80-5aad-3319-094ff69180bb&l=13354

Hi  @aljoscha thanks for your help. As a new hand, actually I don't find the CI website. 

Could you show me the link? thanks a lot

**Comment by twalthr** [comment]
@darionyaphet You can find the instructions how to set up your CI here:
https://cwiki.apache.org/confluence/display/FLINK/Azure+Pipelines
under `https://cwiki.apache.org/confluence/display/FLINK/Azure+Pipelines`

**Review by twalthr — APPROVED** [review]
LGTM will merge

**Comment by aljoscha** [comment]
@darionyaphet You can see the CI report from the post that the `flinkbot` does. Where it says "CI report" you will see "SUCCESS", or "PENDING", or "FAILED". You can click that link to see the logs and see where the problem is.

**Pull request closed without merging**